projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -993,28 +991,16 @@ public function get_filtered_connection_data( $selected_post_id = null ) {
 					$enabled = true;
 				}
 
-				$connection_list[] = array(
-					// REST Meta fields.
-					'connection_id'   => $connection_id,
-					'display_name'    => $this->get_display_name( $service_name, $connection ),
-					'enabled'         => $enabled,
-					'external_handle' => $this->get_external_handle( $service_name, $connection ),
-					'external_id'     => $connection_meta['external_id'] ?? '',
-					'profile_link'    => (string) $this->get_profile_link( $service_name, $connection ),
-					'profile_picture' => (string) $this->get_profile_picture( $connection ),
-					'service_label'   => static::get_service_label( $service_name ),
-					'service_name'    => $service_name,
-					'shared'          => ! $connection_data['user_id'],
-					'status'          => null,
-
-					// Deprecated fields.
-					'id'              => $connection_id,
-					'unique_id'       => $unique_id,
-					'username'        => $this->get_username( $service_name, $connection ),
-					'done'            => $done,
-					'toggleable'      => $toggleable,
-					'global'          => 0 == $connection_data['user_id'], // phpcs:ignore Universal.Operators.StrictComparisons.LooseEqual,WordPress.PHP.StrictComparisons.LooseComparison -- Other types can be used at times.
-					'user_id'         => (int) $connection_data['user_id'],
+				$connection_list[] = array_merge(
+					$connection,
+					array(
+						'enabled'    => $enabled,
+						// Deprecated fields.
+						'id'         => $connection_id,
+						'unique_id'  => $unique_id,
+						'done'       => $done,
+						'toggleable' => $toggleable,
+					)
```

Comment:
I'm not sure if connection contains all the previous fields, but if not, then is it possible that will cause problems with sites running multiple versions of plugins?